sql/hive/src/main/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -66,6 +66,10 @@ class TestHiveContext(sc: SparkContext) extends HiveContext(sc) {
   // without restarting the JVM.
   System.clearProperty("spark.hostPort")
   CommandProcessorFactory.clean(hiveconf)
+  System.setProperty("hive.version", HiveShim.version)
```

Comment:
There is a SerDe jar at sql/hive/src/test/resources/data/files/TestSerDe.jar , Instead of whitelisting the `mapjoin_addjar` in `HiveCompatibilitySuite.scala`, we can create a new unit test in `HiveQuerySuite` for simplifying the unit testing. As setting the local maven repo like this seems a little bit hacky to me, it probably doesn't work if people modify the default location or the jar `hive-hcatalog-core-*.jar` doesn't exist in some cases.